### ISSUE TITLE
Schema violations during ingestion should provide line numbers.

### ISSUE THREAD

**Issue by edwardchapin** [tmtsoftware/icd#45]
When editing model files it is easy to make schema errors, but when ingesting, it is hard to find out what they are due to a lack of line numbers. This issue is relevant to **icd-db 0.9**. See the following examples:

1. Set an invalid variable, e.g., in `publish-model.conf`:
```
bla = 42

$ icd-db -i .
error: ./csro/oiwfs-poa-assembly/publish-model.conf: error: object instance has properties which are not allowed by the schema: ["bla"]
```
2. A common mistake is to specify "int" instead of "integer" for a data type, e.g., for a published event:
```
        {
          name          = trackID1
          description   = "Unique TCS target ID that is incremented (with rollover) each time the TCS is instructed to move to a new target."
          type          = int
        }

$ icd-db -i .
error: ./csro/oiwfs-poa-assembly/publish-model.conf, at path: /publish/events/2/attributes/1: error: instance failed to match all required schemas (matched only 0 out of 1) (schema: config:/attributes-schema.conf#:/items)
instance failed to match at least one required schema among 2
```

**Comment by abrighton** [comment]
This is difficult, since the original HOCON formatted model file is first converted to JSON and then validated, so the original line numbers will have changed. Besides that, the validation software does not provide line numbers, since the JSON is often sent as one line: https://github.com/java-json-tools/json-schema-validator/issues/152.

**Comment by edwardchapin** [comment]
Yes, I see the problem with getting line numbers.

Regardless, I wonder if there is some way to get a hint regarding what part of the file caused the problem? The first example I provided isn't so bad because it at least tells you `bla` is the thing it doesn't recognize (and I can then search for the string in the model file to locate the problem). But in the second example there's really nothing to go by (I would expect something like `int doesn't match allowed values for type`).

**Comment by abrighton** [comment]
Yes, the error messages are not very informative.  However from the error message:
   error: ./csro/oiwfs-poa-assembly/publish-model.conf, at path: /publish/events/2/attributes/1
you know the file and the JSON path: In the "publish" section, the second event (/2), first attribute (/1).

But with a little work, we could probably add a custom validator that could give better errors. 
This would probably have to be hard coded, not just based on the JSON schema.

**Comment by edwardchapin** [comment]
OK, that's actually enough to work things out in a pinch (I hadn't bothered delving into what that "path" actually meant). Probably a note in the README about how to interpret such messages is enough for now.

Another (hopefully easy) suggestion: when it fails, does it actually return a partially-parsed object? i.e., in code, would it be possibly to handle the error and then say something like "the event after [whatever event 1 is, since it parsed correctly] has malformed attribute 1".
